service/worker/batcher/workflow.go

```diff
@@ -235,6 +247,24 @@ func validateParams(params BatchParams) error {
 			return fmt.Errorf("must provide signal name")
 		}
 		return nil
+	case BatchTypeUpdateOptions:
+		if params.UpdateOptionsParams.WorkflowExecutionOptions == nil {
+			return fmt.Errorf("must provide UpdateOptions")
+		}
+		if params.UpdateOptionsParams.UpdateMask == nil {
+			return fmt.Errorf("must provide UpdateMask")
+		}
+		if override := params.UpdateOptionsParams.WorkflowExecutionOptions.VersioningOverride; override != nil {
+			if override.GetBehavior() == enumspb.VERSIONING_BEHAVIOR_PINNED &&
+				override.GetDeployment() == nil {
+				return fmt.Errorf("must provide Deployment if Behavior is 'PINNED'")
+			}
+			if override.GetBehavior() != enumspb.VERSIONING_BEHAVIOR_PINNED &&
+				override.GetDeployment() != nil {
+				return fmt.Errorf("only provide Deployment if Behavior is 'PINNED'")
+			}
```

Comment:
should we add this check to APIs around single mode?

Reply:
I think I already do, but I will make a shared `validate` function and call that

Reply:
done